rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,503 @@
+- Contribution Name: `Alert Manager Integration`
+- Implementation Owner: @christyjacob4
+- Start Date: 2020-03-03
+- Target Date: 
+- RFC PR: [linkerd/gsoc#0000](https://github.com/linkerd/gsoc/pull/0000)
+- *Linkerd* Issue: [linkerd/linkerd2#1726](https://github.com/linkerd/linkerd2/issues/1726)
+- Reviewers: 
+
+# Summary
+
+[summary]: #summary
+
+*Linkerd* uses Prometheus for collecting metrics from various endpoints. Out of the box, Prometheus is an open source monitoring solution that gathers time series based numerical data. Your services need to expose an endpoint (/metrics) from which Prometheus can scrape metrics. The Prometheus monitoring suite comes with tools that help enhance its capabilities like Grafana and AlertManager. The current implementation of the Control Plane makes use of Prometheus for scraping endpoints and aggregating metrics.
+
+The primary goal of this project is to integrate the Alertmanager into the current Control Plane so that Prometheus can provide out of the box alerts to the preferred channels.
+AlertManager can be used for grouping alerts, silencing alerts, routing and sending alerts to preferred channels like slack, emails, pagerduty etc. 
+This will allow users and teams to keep track of critical events that occur on their deployments with ease. 
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+- By default, the *Linkerd* Control Plane doesn't have an alerting mechanism. Prometheus gathers all the metrics which can be visualised through Grafana, but in order to send these alerts to channels, we need to integrate Alertmanager with Prometheus as part of the control plane.
+
+* Once this RFC is implemented, users will have the option to install **Alertmanager** in the control plane using the `addon-config` flag in `linkerd install`
+
+
+- The optional **Alertmanager** installation will come with some default alerts
+  which users can configure to appropriate channels like slack, email, pagerduty, wechat etc.
+
+- Users will also be able to specify custom rules for Prometheus as part of the configuration for `--addon-config` in `config.yaml`.
+
+- Integration with Service Profiles will allow users to define rules and alerts for per route metrics providing extremely fine grain alerting mechanisms for their apps.
+
+A POC for this has been done in [linkerd/linkerd2#1726](https://github.com/linkerd/linkerd2/pull/4124)
+
+# Design proposal (Step 2)
+
+The Integration of Alertmanager with the Existing Control Plane is a multi step process. It can be divided into the following stages
+
+---
+- ## Providing an option to include AlertManager in the installation
+The *Linkerd* install command is currently used to generate a helm chart based on various flags and options that are passed to it. The helm chart is then applied using `kubectl apply` .
```

Comment:
The install command does not generate a helm chart. It just outputs the control plane resource specification in YAML format.